lib/logflare/backends/source_sup.ex

```diff
@@ -50,7 +51,7 @@ defmodule Logflare.Backends.SourceSup do
     children =
       [
         {RateCounterServer, [source: source]},
-        {RecentEventsTouch, [source: source]},
+        {GenSingleton, child_spec: {RecentEventsTouch, source: source}},
```

Comment:
I would consider having a separate Supervisor for  `GenSingleton` with `strategy: :rest_for_one`. This way, if the GenSingleton "watcher" process (the current GenServer) terminates, it would also terminate its child, and you keep their failures contained/related.